lib/states/choices.js

```diff
@@ -403,27 +416,49 @@ var PaginatedChoiceState = ChoiceState.extend(function(self, name, opts) {
         var n = self.options_per_page;
         if (n !== null) return n;
 
-        var chars = self._chars();
         n = self.metadata.page_start;
+        var chars = self._chars(n);
+        var page_choices = 1;
 
         while (n-- > 0) {
-            chars -= self.format_choice(self.choices[n], n).length;
-            if (chars < 0) break;
+            chars -= self.format_choice(self.choices[n], page_choices).length;
+            page_choices++;
+
+            if (chars < 0) {
+                // check if previous choice is the first choice
+                if (n-1 === -1) {
+                    // check if first choice fits in 'Back' space
+                    chars += self.format_choice(self.back, self.choices.length).length;
+                    if (chars >= 0) {
+                        n--;  // decr n if it fits
+                    }
+                }
+                break;
+            }
         }
 
         return self.metadata.page_start - n - 1;
     };
 
-    self._chars = function() {
+    self._chars = function(n) {
         // TODO this needs to be made less naive, it assumes that the last page
         // needs space for a 'more' choice and that the first page needs space
         // for a 'back' choice
-        return self.characters_per_page - [
-            self.format_choice(self.back, self.choices.length),
-            self.format_choice(self.more, self.choices.length),
-            self.question_text + '\n']
-            .join('')
-            .length;
+        if (n === -1) {
+            return self.characters_per_page - [
+                self.format_choice(self.more, self.choices.length),
+                self.question_text + '\n']
+                .join('')
+                .length;
+        } else {
+            return self.characters_per_page - [
+                self.format_choice(self.back, self.choices.length),
+                self.format_choice(self.more, self.choices.length),
+                self.question_text + '\n']
+                .join('')
+                .length;
+        }
```

Comment:
That's what I thought at first, too, but n is based off the metadata.page_start so it didn't really end up working like that.  It's more a case of this function caters for starting at the beginning, and the other functions handling paging backwards and forwards from there. We can chat through it, maybe there's an error in my reasoning.